<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>
    Pipeline tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs
  </title>
<style>
#title{
  background-color: #ec407a;
  color: white;
  padding: 2em;
}
h1 {
  margin-bottom: 0em;
}
#subtitle {
  font-variant: small-caps;
  text-transform: lowercase;
  letter-spacing: 0.2em;
}
.run-info ul {
  list-style: none;
}
.run-info ul li {
  display: inline;
  margin-right: 6em;
  margin-left: -3em;
  text-transform: uppercase;
  font-size: small;
  color: #ec407a;
}
#content {
  max-width: 800px;
  margin-left: auto;
  margin-right: auto;
  padding-top: 2em;
  color: #263238;
  font-family: Helvetica, sans-serif;
}
p {
  margin: 0;
}
.command-table {
  background-color: #eceff1;
  padding: 0.3em;
  border-radius: 0.3em;
  margin-bottom: 4em;
}
.command-table .fail-ignored {
  background-color: #e65100cc;
  padding: 0.3em;
  border-radius: 0.3em;
}
.command-table .success {
  background-color: #1a237ecc;
  padding: 0.3em;
  border-radius: 0.3em;
}
.command-table .in-progress {
  background-color: black;
  padding: 0.3em;
  border-radius: 0.3em;
}
.command-table .non-started {
  background-color: #26323844;
  padding: 0.3em;
  border-radius: 0.3em;
}
.command-table .fail {
  background-color: #f44336cc;
  padding: 0.3em;
  border-radius: 0.3em;
}
.command-content {
  background-color: #eceff1;
  padding: 1em;
}
.command-header {
  display: flex;
  align-items: center;
  margin-bottom: 1em;
}
.command-id-link{
  flex-grow: 0;
}
.command-id-link a {
  font-size: xx-large;
  text-align: center;
  font-weight: bold;
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb;
  text-decoration: none;
  margin-left: 0.2em;
}
.command-id-link a:link {
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb;
}
.command-id-link a:visited {
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb;
}
.command-id-link a:visited:hover {
  color: #ec407a;
  background-color: #00000000;
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb,
               0px 0px 10px #fff,
               0px 0px 5px #ec407a,
               0px 0px 5px #ec407a;
}
.command-id-link a:hover {
  color: #ec407a;
  background-color: #00000000;
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb
               0px 0px 10px #fff,
               0px 0px 5px #ec407a,
               0px 0px 5px #ec407a;
}
.command-id-link a:active {
  color: #ffeb3b;
  background-color: #00000000;
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb,
                 0px 0px 10px #fff,
                 0px 0px 5px #ffeb3b,
                 0px 0px 5px #ffeb3b;
}
.command-id-link a:visited:active {
  color: #ffeb3b;
  background-color: #00000000;
  text-shadow: 1px  1px 0px #000000bb,
              -1px -1px 0px #ffffffbb,
                 0px 0px 10px #fff,
                 0px 0px 5px #ffeb3b,
                 0px 0px 5px #ffeb3b;
}
.command-description {
  color: #26323899;
  font-size: smaller;
  text-align: center;
  font-weight: bold;
  flex-grow: 1;
}

.command-invocation {
  padding: 1em;
  color: #fff;
  background-color: #005662;
  font-size: small;
  font-family: monospace;
  border-radius: 0.5em;
  text-shadow: 0 0 2px #ffccbccc;
}
.command-stats-table {
  margin-top: 0.2em;
  border-radius: 0.5em;
  padding: 0.2em;
  padding-top: 0.4em;
  display: flex;
  flex-direction: column;
  background-color: #eae3998a;
  color: #263238cc;
  margin-bottom: 0.2em;
}
.command-stats-row {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  margin-top: 0.4em;
  margin-bottom: 0.4em;
  font-size: small;
  text-transform: uppercase;
}
.command-stats-row p .value {
  color: #00251aaa;
  text-shadow: 0 0 2px #82ada944;
  font-weight: bold;
}
.literal-stdout {
  margin-top: 0.4em;
  margin-bottom: 0.2em;
  padding: 1em;
  border-radius: 0.5em;
  overflow-x: auto;
  overflow-y: auto;
}
.output-box {
  margin-top: 0.4em;
  margin-bottom: 0.2em;
  padding: 1em;
  color: #000a12;
  background-color: #cfd8dc;
  font-size: small;
  font-family: monospace;
  border-radius: 0.5em;
  max-height: 20em;
  overflow-x: auto;
  overflow-y: auto;
}
.output-box xmp{
  white-space: pre;
}
.stage-name p {
  margin-left: 0.5em;
  color: #546e7a;
  text-shadow:  -1px  0px 0px #fff
               ,-2px  0px 0px #ffffff99
               ,-1px  1px 0px #ffffff99
               ;
  font-size: 48pt;
  font-variant: small-caps;
  text-transform: lowercase;
  font-weight: bold;
  font-style: italic;
}
.stage-name .underline {
  width: 100%;
  height: 1.1em;
  background-color: #546e7a;
  margin-top: -1.6em;
}
.stage-name {
  margin-top: 6em;
  margin-bottom: 3em;
}

.outcome-table-info-box {
  margin-top: 0.4em;
  margin-bottom: 0.2em;
  padding: 1em;
  color: #000a12;
  background-color: #ffee58;
  border-radius: 0.5em;
}
.outcome-table-info-box .outcome{
  font-weight: bold;
}
#toc {
  margin-bottom: 6em;
  display: flex;
  flex-direction: column;
  flex-shrink: 1;
  margin-right: 3em;
}
.toc-title {
  text-align: center;
  font-variant: small-caps;
  text-transform: lowercase;
  margin-bottom: -1em;
}
.toc-section{
  margin-top: 1em;
  margin-bottom: 0.5em;
  text-transform: capitalize;
}
.toc-row {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}
.toc-row-icon {
  padding-right: 8px;
}
.status-icon-cross {
  fill: #f44336;
    width: 16px;
    height: 16px;
}
.status-icon-tick {
  fill: #3f51b5;
  stroke: #ffffff77;
  stroke-width: 1px;
  width: 16px;
  height: 16px;
}

.status-icon-in-progress {
  fill: #333;
  width: 16px;
  height: 16px;
}

.status-icon-in-progress circle {
  stroke: #333;
  stroke-width: 2px;
  fill: none
}
a:link {
  color: #1e88e5;
}
a:visited {
  color: #1e88e5;
}
a:visited:hover {
  background-color: #1e88e5;
  color: #fff;
  text-decoration: none;
}
a:hover {
  background-color: #1e88e5;
  color: #fff;
  text-decoration: none;
}
a:active {
  background-color: #ec407a;
  color: #fff;
  text-decoration: none;
}
a:visited:active {
  background-color: #ec407a;
  color: #fff;
  text-decoration: none;
}

#contents-container {
  margin-top: 6em;
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  align-content: flex-start;
}

#depgraph {
  display: flex;
  flex-direction: column;
  align-items: center;
  align-content: flex-start;
  flex-shrink: 7;
}

#depgraph-title p {
  text-align: center;
  font-variant: small-caps;
  text-transform: lowercase;
}

#depgraph-preview img {
  max-width: 100%;
  max-height: 100%;
}

#depgraph-preview {
  padding: 1em;
}

.memory-box {
  display: flex;
  justify-content: space-between;
  padding: 0.5em;
  align-items: center;
}
.memory-preview {
  width: 75%;
}
.memory-preview svg {
  margin-left: -1em;
  margin-right: -2em;
}
.memory-table {
}
.memory-title {
  text-align: center;
  font-style: bold;
}
.memory-subtitle {
  text-transform: lowercase;
  font-variant: small-caps;
  text-align: center;
  opacity: 0.8;
}
.memory-pad {
  width: 1.5em;
}
.memory-value {
  text-align: center;
  margin-right: 2em;
}
.rss {
  color: #ab47bc;
}
.vsz {
  color: #4caf50;
}


@media (prefers-color-scheme: dark){
  a:link {
    color: #29b6f6;
  }
  a:visited {
    color: #29b6f6;
  }
  a:visited:hover {
    color: #263238;
    background-color: #29b6f6;
  }
  a:hover {
    color: #263238;
    background-color: #29b6f6;
  }
  a:active {
    color: #263238;
    background-color: #ec407a;
  }
  a:visited:active {
    color: #263238;
    background-color: #ec407a;
  }
  .command-id-link a:link {
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb;
  }
  .command-id-link a:visited {
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb;
  }
  .command-id-link a:visited:hover {
    color: #ec407a;
    background-color: #00000000;
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb.
                 0px 0px 10px #fff,
                 0px 0px 5px #ec407a,
                 0px 0px 5px #ec407a;
  }
  .command-id-link a:hover {
    color: #ec407a;
    background-color: #00000000;
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb,
                 0px 0px 10px #fff,
                 0px 0px 5px #ec407a,
                 0px 0px 5px #ec407a;
  }
  .command-id-link a:active {
    color: #ffeb3b;
    background-color: #00000000;
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb,
                 0px 0px 10px #fff,
                 0px 0px 5px #ffeb3b,
                 0px 0px 5px #ffeb3b;
  }
  .command-id-link a:visited:active {
    color: #ffeb3b;
    background-color: #00000000;
    text-shadow: 1px  1px 0px #000000bb,
                -1px -1px 0px #ffffffbb,
                 0px 0px 10px #fff,
                 0px 0px 5px #ffeb3b,
                 0px 0px 5px #ffeb3b;
  }
  body {
    color: #babdbe;
    background-color: #263238;
  }
  .outcome-table-info-box p{
    color: #000a12;
  }
  p {
    color: #babdbe;
  }
  #subtitle {
    color: #fff;
  }
  .command-table {
    background-color: #37474f;
  }
  .command-content {
    background-color: #37474f;
  }
  .command-invocation p {
    color: #fff;
  }
  .command-stats-table {
    background-color: #eae399d0;
    color: #263238;
  }
  .command-stats-table p {
    color: #263238;
  }
  .output-box {
    color: #cfd8dccc;
    background-color: #102027;
  }
  .stage-name p {
    text-shadow: -1px  0px 0px #263238,
                 -2px  0px 0px #26323899,
                 -1px  1px 0px #26323899;
  }
  .status-icon-in-progress { fill: #ddd; }
  .status-icon-in-progress circle { stroke: #ddd; }

  /* SVG graphs */
  g text {
    fill: #eceff1;
    stroke: #fff0;
  }
}
</style>
</head>
<body>
<div id="content">

  <div id="title">
    <h1>
      Book Runner pipeline &lsquo;tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs&rsquo;
    </h1>
    <p id="subtitle">
      Litani CI Dashboard
    </p>
  </div><!-- id="title" -->

  <div class=run-info>
    <ul>
      <li>Pipeline status: success</li>
      <li>2023-12-22T09:45:14Z</li>
      <li>ab447272-4adb-4e04-a944-4db64b0a01b0</li>
    </ul>
  </div><!-- class=run-info -->

  <div id="contents-container">

    <div id="toc">
      <div class="toc-title">
        <p>Summary of jobs for this pipeline</p>
      </div><!-- class="toc-title" -->

      
        <div class="toc-section">
          <p>check</p>
        </div><!-- class="toc-section" -->
        
          <div class="toc-row">

            <div class="toc-row-icon">
              
                <svg height="16px" width="16px" class="status-icon-tick">
                  <polygon points="1,11 6,15 15,3 13,1 6,12 3,8"></polygon>
                </svg>
              
            </div><!-- class="toc-row-icon" -->

            <div class="toc-row-job-name">
              <p>
              <a href="#job-0e2b8280-6ff2-4031-bf3b-20b401677b79">
              
              Is this valid Rust code?
              
              </a>
              </p>
            </div><!-- class="toc-row-job-name" -->

          </div><!-- class="toc-row" -->

        
      
        <div class="toc-section">
          <p>codegen</p>
        </div><!-- class="toc-section" -->
        
          <div class="toc-row">

            <div class="toc-row-icon">
              
                <svg height="16px" width="16px" class="status-icon-tick">
                  <polygon points="1,11 6,15 15,3 13,1 6,12 3,8"></polygon>
                </svg>
              
            </div><!-- class="toc-row-icon" -->

            <div class="toc-row-job-name">
              <p>
              <a href="#job-7f20933d-fd90-4bda-9f87-c32c6fe6f823">
              
              Does Kani support all the Rust features used in it?
              
              </a>
              </p>
            </div><!-- class="toc-row-job-name" -->

          </div><!-- class="toc-row" -->

        
      
        <div class="toc-section">
          <p>verification</p>
        </div><!-- class="toc-section" -->
        
          <div class="toc-row">

            <div class="toc-row-icon">
              
                <svg height="16px" width="16px" class="status-icon-tick">
                  <polygon points="1,11 6,15 15,3 13,1 6,12 3,8"></polygon>
                </svg>
              
            </div><!-- class="toc-row-icon" -->

            <div class="toc-row-job-name">
              <p>
              <a href="#job-91a86c61-104e-461a-b23a-ebc44d1bf1b4">
              
              Can Kani reason about it?
              
              </a>
              </p>
            </div><!-- class="toc-row-job-name" -->

          </div><!-- class="toc-row" -->

        
      
    </div><!-- id="toc" -->


    

  </div><!-- id="contents-container" -->


  
  <div class="stage">
    <div class="stage-name">
      <p>check</p>
      <div class="underline"></div>
    </div><!-- class="stage-name" -->

    
    <div class="command-table" id="job-0e2b8280-6ff2-4031-bf3b-20b401677b79">
      
      <div class="success">
      
        <div class="command-content">
          <div class="command-header">
            <div class="command-id-link">
              <p>
                <a href="#job-0e2b8280-6ff2-4031-bf3b-20b401677b79">#</a>
              </p>
            </div><!-- class="command-id-link" -->
            <div class="command-description">
              
              <p>
                Is this valid Rust code?
              </p>
              
            </div><!-- class="command-description" -->
          </div><!-- class="command-header" -->
          <div class="command-invocation">
            <p>
            "kani-compiler" "--edition" "2015" "-Z" "no-codegen" "tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs"
            </p>
          </div><!-- class="command-invocation" -->

          
          <div class="command-stats-table">
            
            <div class="command-stats-row">
              <p>Command return code:
              <span class="value">
                0
              </span></p>
              <p>OK returns:
              <span class="value">
                ['0']
              </span></p>
              <p>Ignored returns:
              <span class="value">
                None
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Timeout:
              <span class="value">
                5
              </span></p>
              <p>Timeout ok:
              <span class="value">
                False
              </span></p>
              <p>Timeout ignored:
              <span class="value">
                False
              </span></p>
              <p>Timeout reached:
              <span class="value">
                False
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Start time:
              <span class="value">
                2023-12-22T09:46:55Z
              </span></p>
              <p>Duration:
              <span class="value">
                00s
              </span></p>
              <p>End time:
              <span class="value">
                2023-12-22T09:46:55Z
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Command successful:
              <span class="value">
              
              yes
              
              </span></p>
            </div><!-- class="command-stats-row" -->
             
          </div><!-- class="command-stats-table" -->
          

          


          


          

          
          <div class="output-box"><xmp>stderr:
            
warning: this method call resolves to `<&[T; N] as IntoIterator>::into_iter` (due to backwards compatibility), but will resolve to <[T; N] as IntoIterator>::into_iter in Rust 2021
--> tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:47
|
26 |     println!("Find 2 in array2: {:?}", array2.into_iter().find(|&&x| x == 2));
|                                               ^^^^^^^^^
|
= warning: this changes meaning in Rust 2021
= note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/IntoIterator-for-arrays.html>
= note: `#[warn(array_into_iter)]` on by default
help: use `.iter()` instead of `.into_iter()` to avoid ambiguity
|
26 |     println!("Find 2 in array2: {:?}", array2.iter().find(|&&x| x == 2));
|                                               ~~~~
help: or use `IntoIterator::into_iter(..)` instead of `.into_iter()` to explicitly iterate by value
|
26 |     println!("Find 2 in array2: {:?}", IntoIterator::into_iter(array2).find(|&&x| x == 2));
|                                        ++++++++++++++++++++++++      ~

warning: 1 warning emitted

            </xmp>
          </div><!-- class="output-box" -->
          

        </div><!-- class="command-content" -->
      </div><!-- class="fail" -->
    </div><!-- id="command-table" -->
    

  </div><!-- class="stage" -->
  
  <div class="stage">
    <div class="stage-name">
      <p>codegen</p>
      <div class="underline"></div>
    </div><!-- class="stage-name" -->

    
    <div class="command-table" id="job-7f20933d-fd90-4bda-9f87-c32c6fe6f823">
      
      <div class="success">
      
        <div class="command-content">
          <div class="command-header">
            <div class="command-id-link">
              <p>
                <a href="#job-7f20933d-fd90-4bda-9f87-c32c6fe6f823">#</a>
              </p>
            </div><!-- class="command-id-link" -->
            <div class="command-description">
              
              <p>
                Does Kani support all the Rust features used in it?
              </p>
              
            </div><!-- class="command-description" -->
          </div><!-- class="command-header" -->
          <div class="command-invocation">
            <p>
            "kani-compiler" "--edition" "2015" "--out-dir" "build/tmp" "tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs"
            </p>
          </div><!-- class="command-invocation" -->

          
          <div class="command-stats-table">
            
            <div class="command-stats-row">
              <p>Command return code:
              <span class="value">
                0
              </span></p>
              <p>OK returns:
              <span class="value">
                ['0']
              </span></p>
              <p>Ignored returns:
              <span class="value">
                None
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Timeout:
              <span class="value">
                10
              </span></p>
              <p>Timeout ok:
              <span class="value">
                False
              </span></p>
              <p>Timeout ignored:
              <span class="value">
                False
              </span></p>
              <p>Timeout reached:
              <span class="value">
                False
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Start time:
              <span class="value">
                2023-12-22T09:46:55Z
              </span></p>
              <p>Duration:
              <span class="value">
                01s
              </span></p>
              <p>End time:
              <span class="value">
                2023-12-22T09:46:56Z
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Command successful:
              <span class="value">
              
              yes
              
              </span></p>
            </div><!-- class="command-stats-row" -->
             
          </div><!-- class="command-stats-table" -->
          

          


          


          

          
          <div class="output-box"><xmp>stderr:
            
warning: this method call resolves to `<&[T; N] as IntoIterator>::into_iter` (due to backwards compatibility), but will resolve to <[T; N] as IntoIterator>::into_iter in Rust 2021
--> tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:47
|
26 |     println!("Find 2 in array2: {:?}", array2.into_iter().find(|&&x| x == 2));
|                                               ^^^^^^^^^
|
= warning: this changes meaning in Rust 2021
= note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/IntoIterator-for-arrays.html>
= note: `#[warn(array_into_iter)]` on by default
help: use `.iter()` instead of `.into_iter()` to avoid ambiguity
|
26 |     println!("Find 2 in array2: {:?}", array2.iter().find(|&&x| x == 2));
|                                               ~~~~
help: or use `IntoIterator::into_iter(..)` instead of `.into_iter()` to explicitly iterate by value
|
26 |     println!("Find 2 in array2: {:?}", IntoIterator::into_iter(array2).find(|&&x| x == 2));
|                                        ++++++++++++++++++++++++      ~

warning: 1 warning emitted

            </xmp>
          </div><!-- class="output-box" -->
          

        </div><!-- class="command-content" -->
      </div><!-- class="fail" -->
    </div><!-- id="command-table" -->
    

  </div><!-- class="stage" -->
  
  <div class="stage">
    <div class="stage-name">
      <p>verification</p>
      <div class="underline"></div>
    </div><!-- class="stage-name" -->

    
    <div class="command-table" id="job-91a86c61-104e-461a-b23a-ebc44d1bf1b4">
      
      <div class="success">
      
        <div class="command-content">
          <div class="command-header">
            <div class="command-id-link">
              <p>
                <a href="#job-91a86c61-104e-461a-b23a-ebc44d1bf1b4">#</a>
              </p>
            </div><!-- class="command-id-link" -->
            <div class="command-description">
              
              <p>
                Can Kani reason about it?
              </p>
              
            </div><!-- class="command-description" -->
          </div><!-- class="command-header" -->
          <div class="command-invocation">
            <p>
            RUSTFLAGS="--edition 2015" RUSTFLAGS="--edition 2015" "kani" "tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs" "--enable-unstable" "--function" "main" "--enable-unstable" "--cbmc-args" "--unwind" "4"
            </p>
          </div><!-- class="command-invocation" -->

          
          <div class="command-stats-table">
            
            <div class="command-stats-row">
              <p>Command return code:
              <span class="value">
                0
              </span></p>
              <p>OK returns:
              <span class="value">
                ['0']
              </span></p>
              <p>Ignored returns:
              <span class="value">
                None
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Timeout:
              <span class="value">
                60
              </span></p>
              <p>Timeout ok:
              <span class="value">
                False
              </span></p>
              <p>Timeout ignored:
              <span class="value">
                False
              </span></p>
              <p>Timeout reached:
              <span class="value">
                False
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Start time:
              <span class="value">
                2023-12-22T09:52:08Z
              </span></p>
              <p>Duration:
              <span class="value">
                11s
              </span></p>
              <p>End time:
              <span class="value">
                2023-12-22T09:52:19Z
              </span></p>
            </div><!-- class="command-stats-row" -->
            <div class="command-stats-row">
              <p>Command successful:
              <span class="value">
              
              yes
              
              </span></p>
            </div><!-- class="command-stats-row" -->
             
          </div><!-- class="command-stats-table" -->
          

          


          


          
            
            <div class="output-box"><xmp>stdout:
              
Kani Rust Verifier 0.43.0 (standalone)
Checking harness main...
CBMC 5.95.1 (cbmc-5.95.1)
CBMC version 5.95.1 (cbmc-5.95.1) 64-bit x86_64 linux
Reading GOTO program from file /home/runner/work/kani/kani/tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.out
Generating GOTO Program
Adding CPROVER library (x86_64)
Removal of function pointers and virtual functions
Generic Property Instrumentation
Running with 16 object bits, 48 offset bits (user-specified)
Starting Bounded Model Checking
Unwinding loop _RINvXs2T_NtNtCseojaShRnPnU_4core5slice4iterINtB7_4IterlENtNtNtNtBb_4iter6traits8iterator8Iterator4findNCNvCsde7GhDWuKEy_2_224main0EB1G_.0 iteration 1 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs line 302 column 24 function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:53: 15:58}> thread 0
Unwinding loop _RINvYINtNtNtCs5ZRniFFnfY1_5alloc3vec9into_iter8IntoIterlENtNtNtNtCseojaShRnPnU_4core4iter6traits8iterator8Iterator8try_folduNCINvNvBT_4find5checklNCNvCsde7GhDWuKEy_2_224mains_0E0INtNtNtB11_3ops12control_flow11ControlFlowlEEB2o_.0 iteration 1 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs line 2461 column 9 function <std::vec::IntoIter<i32> as std::iter::Iterator>::try_fold::<(), {closure@std::iter::Iterator::find::check<i32, {closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:53: 18:58}>::{closure#0}}, std::ops::ControlFlow<i32>> thread 0
Unwinding loop _RINvYINtNtNtCs5ZRniFFnfY1_5alloc3vec9into_iter8IntoIterlENtNtNtNtCseojaShRnPnU_4core4iter6traits8iterator8Iterator8try_folduNCINvNvBT_4find5checklNCNvCsde7GhDWuKEy_2_224mains_0E0INtNtNtB11_3ops12control_flow11ControlFlowlEEB2o_.0 iteration 2 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs line 2461 column 9 function <std::vec::IntoIter<i32> as std::iter::Iterator>::try_fold::<(), {closure@std::iter::Iterator::find::check<i32, {closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:53: 18:58}>::{closure#0}}, std::ops::ControlFlow<i32>> thread 0
Unwinding loop _RINvYINtNtNtCs5ZRniFFnfY1_5alloc3vec9into_iter8IntoIterlENtNtNtNtCseojaShRnPnU_4core4iter6traits8iterator8Iterator8try_folduNCINvNvBT_4find5checklNCNvCsde7GhDWuKEy_2_224mains_0E0INtNtNtB11_3ops12control_flow11ControlFlowlEEB2o_.0 iteration 3 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs line 2461 column 9 function <std::vec::IntoIter<i32> as std::iter::Iterator>::try_fold::<(), {closure@std::iter::Iterator::find::check<i32, {closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:53: 18:58}>::{closure#0}}, std::ops::ControlFlow<i32>> thread 0
Unwinding loop _RINvXs2T_NtNtCseojaShRnPnU_4core5slice4iterINtB7_4IterlENtNtNtNtBb_4iter6traits8iterator8Iterator4findNCNvCsde7GhDWuKEy_2_224mains0_0EB1G_.0 iteration 1 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs line 302 column 24 function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:64: 24:69}> thread 0
Unwinding loop _RINvXs2T_NtNtCseojaShRnPnU_4core5slice4iterINtB7_4IterlENtNtNtNtBb_4iter6traits8iterator8Iterator4findNCNvCsde7GhDWuKEy_2_224mains1_0EB1G_.0 iteration 1 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs line 302 column 24 function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}> thread 0
Unwinding loop _RINvXs2T_NtNtCseojaShRnPnU_4core5slice4iterINtB7_4IterlENtNtNtNtBb_4iter6traits8iterator8Iterator4findNCNvCsde7GhDWuKEy_2_224mains1_0EB1G_.0 iteration 2 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs line 302 column 24 function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}> thread 0
Unwinding loop _RINvXs2T_NtNtCseojaShRnPnU_4core5slice4iterINtB7_4IterlENtNtNtNtBb_4iter6traits8iterator8Iterator4findNCNvCsde7GhDWuKEy_2_224mains1_0EB1G_.0 iteration 3 file /home/runner/.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs line 302 column 24 function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}> thread 0
Runtime Symex: 1.00807s
size of program expression: 14003 steps
slicing removed 9045 assignments
Generated 829 VCC(s), 177 remaining after simplification
Runtime Postprocess Equation: 0.00811414s
Passing problem to propositional reduction
converting SSA
Runtime Convert SSA: 0.096555s
Running propositional reduction
Post-processing
Runtime Post-process: 0.00011196s
Solving with CaDiCaL 1.7.2
24750 variables, 25925 clauses
SAT checker: instance is SATISFIABLE
Runtime Solver: 0.00221454s
Runtime decision procedure: 0.113308s
Running propositional reduction
Solving with CaDiCaL 1.7.2
24750 variables, 25926 clauses
SAT checker: instance is UNSATISFIABLE
Runtime Solver: 0.00189226s
Runtime decision procedure: 0.00244298s

RESULTS:
Check 1: main.assertion.1
- Status: SUCCESS
- Description: "misaligned pointer dereference: address must be a multiple of its type's alignment"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 2: main.assertion.2
- Status: SUCCESS
- Description: "misaligned pointer dereference: address must be a multiple of its type's alignment"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 3: __rust_dealloc.assertion.1
- Status: SUCCESS
- Description: "Alignment is power of two"
- Location: library/kani/kani_lib.c:84 in function __rust_dealloc

Check 4: __rust_dealloc.assertion.2
- Status: SUCCESS
- Description: "rust_dealloc must be called on an object whose allocated size matches its layout"
- Location: library/kani/kani_lib.c:86 in function __rust_dealloc

Check 5: malloc.assertion.1
- Status: SUCCESS
- Description: "max allocation size exceeded"
- Location: <builtin-library-malloc>:31 in function malloc

Check 6: malloc.assertion.2
- Status: SUCCESS
- Description: "max allocation may fail"
- Location: <builtin-library-malloc>:36 in function malloc

Check 7: std::panic::Location::<'_>::caller.unsupported_construct.1
- Status: SUCCESS
- Description: "caller_location is not currently supported by Kani. Please post your example at https://github.com/model-checking/kani/issues/374"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic/location.rs:88:9 in function std::panic::Location::<'_>::caller

Check 8: core::panicking::panic_nounwind_fmt::runtime.assertion.1
- Status: SUCCESS
- Description: "reached intrinsic::abort"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:91:13 in function core::panicking::panic_nounwind_fmt::runtime

Check 9: std::ptr::NonNull::<i32>::add.arithmetic_overflow.1
- Status: SUCCESS
- Description: "offset: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:623:37 in function std::ptr::NonNull::<i32>::add

Check 10: std::ptr::NonNull::<i32>::add.arithmetic_overflow.2
- Status: SUCCESS
- Description: "attempt to compute offset which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:623:37 in function std::ptr::NonNull::<i32>::add

Check 11: std::ptr::const_ptr::<impl *const std::alloc::Global>::is_aligned_to.assertion.1
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:1607:13 in function std::ptr::const_ptr::<impl *const std::alloc::Global>::is_aligned_to

Check 12: core::panicking::assert_failed_inner.assertion.1
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:342:17 in function core::panicking::assert_failed_inner

Check 13: core::panicking::assert_failed_inner.assertion.2
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:337:23 in function core::panicking::assert_failed_inner

Check 14: core::panicking::assert_failed_inner.unreachable.1
- Status: SUCCESS
- Description: "unreachable code"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:330:20 in function core::panicking::assert_failed_inner

Check 15: std::fmt::Arguments::<'_>::new_v1.assertion.1
- Status: SUCCESS
- Description: "attempt to add with overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:332:56 in function std::fmt::Arguments::<'_>::new_v1

Check 16: std::fmt::Arguments::<'_>::new_v1.assertion.2
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:333:13 in function std::fmt::Arguments::<'_>::new_v1

Check 17: std::ptr::const_ptr::<impl *const i32>::is_aligned_to.assertion.1
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:1607:13 in function std::ptr::const_ptr::<impl *const i32>::is_aligned_to

Check 18: <std::vec::IntoIter<i32> as std::iter::Iterator>::try_fold::<(), {closure@std::iter::Iterator::find::check<i32, {closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:53: 18:58}>::{closure#0}}, std::ops::ControlFlow<i32>>.unreachable.1
- Status: SUCCESS
- Description: "unreachable code"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/iterator.rs:2462:21 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::try_fold::<(), {closure@std::iter::Iterator::find::check<i32, {closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:53: 18:58}>::{closure#0}}, std::ops::ControlFlow<i32>>

Check 19: <std::ops::ControlFlow<i32> as std::ops::Try>::branch.unreachable.1
- Status: SUCCESS
- Description: "unreachable code"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/control_flow.rs:111:15 in function <std::ops::ControlFlow<i32> as std::ops::Try>::branch

Check 20: std::ptr::const_ptr::<impl *const i32>::add.arithmetic_overflow.1
- Status: SUCCESS
- Description: "offset: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:944:18 in function std::ptr::const_ptr::<impl *const i32>::add

Check 21: std::ptr::const_ptr::<impl *const i32>::add.arithmetic_overflow.2
- Status: SUCCESS
- Description: "attempt to compute offset which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:944:18 in function std::ptr::const_ptr::<impl *const i32>::add

Check 22: std::ptr::const_ptr::<impl *const T>::is_aligned_to::runtime_impl.assertion.1
- Status: UNREACHABLE
- Description: "attempt to subtract with overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:1612:26 in function std::ptr::const_ptr::<impl *const T>::is_aligned_to::runtime_impl

Check 23: std::rt::panic_fmt.unsupported_construct.1
- Status: SUCCESS
- Description: "call to foreign "Rust" function `rust_begin_unwind` is not currently supported by Kani. Please post your example at https://github.com/model-checking/kani/issues/new/choose"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:61:9 in function std::rt::panic_fmt

Check 24: core::num::<impl usize>::unchecked_mul.arithmetic_overflow.1
- Status: SUCCESS
- Description: "attempt to compute `unchecked_mul` which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/num/uint_macros.rs:604:22 in function core::num::<impl usize>::unchecked_mul

Check 25: std::ops::ControlFlow::<i32>::break_value.unreachable.1
- Status: SUCCESS
- Description: "unreachable code"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/control_flow.rs:181:15 in function std::ops::ControlFlow::<i32>::break_value

Check 26: std::ptr::mut_ptr::<impl *mut i32>::add.arithmetic_overflow.1
- Status: SUCCESS
- Description: "offset: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mut_ptr.rs:1045:18 in function std::ptr::mut_ptr::<impl *mut i32>::add

Check 27: std::ptr::mut_ptr::<impl *mut i32>::add.arithmetic_overflow.2
- Status: SUCCESS
- Description: "attempt to compute offset which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mut_ptr.rs:1045:18 in function std::ptr::mut_ptr::<impl *mut i32>::add

Check 28: std::intrinsics::write_bytes::<i32>.safety_check.1
- Status: SUCCESS
- Description: "`dst` must be properly aligned"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/intrinsics.rs:2867:9 in function std::intrinsics::write_bytes::<i32>

Check 29: std::intrinsics::write_bytes::<i32>.arithmetic_overflow.1
- Status: SUCCESS
- Description: "write_bytes: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/intrinsics.rs:2867:9 in function std::intrinsics::write_bytes::<i32>

Check 30: std::ptr::const_ptr::<impl *const u8>::wrapping_offset.arithmetic_overflow.1
- Status: SUCCESS
- Description: "arith_offset: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:550:18 in function std::ptr::const_ptr::<impl *const u8>::wrapping_offset

Check 31: std::ptr::const_ptr::<impl *const u8>::wrapping_offset.arithmetic_overflow.2
- Status: SUCCESS
- Description: "attempt to compute offset which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:550:18 in function std::ptr::const_ptr::<impl *const u8>::wrapping_offset

Check 32: std::ptr::const_ptr::<impl *const i32>::sub_ptr.assertion.1
- Status: SUCCESS
- Description: "assertion failed: 0 < pointee_size && pointee_size <= isize::MAX as usize"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:820:9 in function std::ptr::const_ptr::<impl *const i32>::sub_ptr

Check 33: std::ptr::const_ptr::<impl *const i32>::sub_ptr.arithmetic_overflow.1
- Status: SUCCESS
- Description: "attempt to compute offset in bytes which would overflow an `isize`"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:822:18 in function std::ptr::const_ptr::<impl *const i32>::sub_ptr

Check 34: std::ptr::const_ptr::<impl *const i32>::sub_ptr.safety_check.1
- Status: SUCCESS
- Description: "attempt to compute unsigned offset with negative distance"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:822:18 in function std::ptr::const_ptr::<impl *const i32>::sub_ptr

Check 35: core::num::<impl usize>::unchecked_sub.arithmetic_overflow.1
- Status: SUCCESS
- Description: "attempt to compute `unchecked_sub` which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/num/uint_macros.rs:557:22 in function core::num::<impl usize>::unchecked_sub

Check 36: std::ptr::mut_ptr::<impl *mut u8>::wrapping_offset.arithmetic_overflow.1
- Status: SUCCESS
- Description: "arith_offset: attempt to compute number in bytes which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mut_ptr.rs:564:18 in function std::ptr::mut_ptr::<impl *mut u8>::wrapping_offset

Check 37: std::ptr::mut_ptr::<impl *mut u8>::wrapping_offset.arithmetic_overflow.2
- Status: SUCCESS
- Description: "attempt to compute offset which would overflow"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mut_ptr.rs:564:18 in function std::ptr::mut_ptr::<impl *mut u8>::wrapping_offset

Check 38: std::ptr::Alignment::as_usize.assume.1
- Status: SUCCESS
- Description: "assumption failed"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/alignment.rs:94:9 in function std::ptr::Alignment::as_usize

Check 39: std::fmt::Arguments::<'_>::new_const.assertion.1
- Status: SUCCESS
- Description: "This is a placeholder message; Kani doesn't support message formatted at runtime"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:323:13 in function std::fmt::Arguments::<'_>::new_const

Check 40: main.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 41: main.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 42: main.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 43: main.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 44: main.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 45: main.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 46: main.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 47: main.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 48: main.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 49: main.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 50: main.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 51: main.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/macros.rs:54:37 in function main

Check 52: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 53: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 54: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 55: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 56: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 57: <std::vec::Vec<i32> as std::ops::Drop>::drop.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:3108:81 in function <std::vec::Vec<i32> as std::ops::Drop>::drop

Check 58: memcmp.precondition.1
- Status: SUCCESS
- Description: "memcmp region 1 readable"
- Location: <builtin-library-memcmp>:19 in function memcmp

Check 59: memcmp.precondition.2
- Status: SUCCESS
- Description: "memcpy region 2 readable"
- Location: <builtin-library-memcmp>:21 in function memcmp

Check 60: memcmp.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 61: memcmp.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 62: memcmp.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 63: memcmp.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 64: memcmp.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 65: memcmp.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 66: memcmp.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 67: memcmp.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 68: memcmp.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 69: memcmp.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 70: memcmp.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 71: memcmp.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: <builtin-library-memcmp>:27 in function memcmp

Check 72: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 73: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 74: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 75: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 76: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 77: std::vec::IntoIter::<i32>::as_raw_mut_slice.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:100:39 in function std::vec::IntoIter::<i32>::as_raw_mut_slice

Check 78: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:301:27 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}>

Check 79: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:306:17 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:64: 26:69}>

Check 80: __rust_dealloc.precondition_instance.1
- Status: SUCCESS
- Description: "free argument must be NULL or valid pointer"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 81: __rust_dealloc.precondition_instance.2
- Status: SUCCESS
- Description: "free argument must be dynamic object"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 82: __rust_dealloc.precondition_instance.3
- Status: SUCCESS
- Description: "free argument has offset zero"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 83: __rust_dealloc.precondition_instance.4
- Status: SUCCESS
- Description: "double free"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 84: __rust_dealloc.precondition_instance.5
- Status: SUCCESS
- Description: "free called for new[] object"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 85: __rust_dealloc.precondition_instance.6
- Status: SUCCESS
- Description: "free called for stack-allocated object"
- Location: library/kani/kani_lib.c:88 in function __rust_dealloc

Check 86: <alloc::raw_vec::RawVec<i32> as std::ops::Drop>::drop.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:504:16 in function <alloc::raw_vec::RawVec<i32> as std::ops::Drop>::drop

Check 87: std::intrinsics::write_bytes::<i32>.precondition_instance.1
- Status: SUCCESS
- Description: "memset destination region writeable"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/intrinsics.rs:2867:9 in function std::intrinsics::write_bytes::<i32>

Check 88: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 89: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 90: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 91: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 92: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 93: alloc::raw_vec::RawVec::<i32>::ptr.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:223:9 in function alloc::raw_vec::RawVec::<i32>::ptr

Check 94: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 95: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 96: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 97: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 98: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 99: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:22 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 100: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 101: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 102: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 103: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 104: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 105: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:212:39 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 106: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 107: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 108: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 109: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 110: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 111: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 112: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.19
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 113: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.20
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 114: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.21
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 115: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.22
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 116: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.23
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 117: <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint.pointer_dereference.24
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:210:42 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::size_hint

Check 118: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 119: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 120: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 121: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 122: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 123: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:12 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 124: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 125: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 126: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 127: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 128: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 129: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:190:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 130: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 131: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 132: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 133: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 134: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 135: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:200:23 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 136: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.19
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 137: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.20
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 138: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.21
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 139: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.22
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 140: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.23
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 141: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.24
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:201:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 142: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.25
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 143: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.26
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 144: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.27
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 145: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.28
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 146: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.29
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 147: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.30
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:24 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 148: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.31
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 149: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.32
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 150: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.33
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 151: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.34
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 152: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.35
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 153: <std::vec::IntoIter<i32> as std::iter::Iterator>::next.pointer_dereference.36
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:195:13 in function <std::vec::IntoIter<i32> as std::iter::Iterator>::next

Check 154: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 155: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 156: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 157: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 158: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 159: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:9 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 160: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 161: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 162: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 163: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 164: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 165: <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:1796:26 in function <std::ptr::NonNull<i32> as std::cmp::PartialEq>::eq

Check 166: std::fmt::Arguments::<'_>::new_v1.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:335:34 in function std::fmt::Arguments::<'_>::new_v1

Check 167: <std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/macros/mod.rs:45:85 in function <std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 168: core::panicking::assert_failed_inner.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:336:11 in function core::panicking::assert_failed_inner

Check 169: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 170: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 171: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 172: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 173: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 174: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:102:27 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 175: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 176: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 177: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 178: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 179: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 180: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:44 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 181: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 182: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 183: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 184: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 185: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 186: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:109:33 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 187: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.19
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 188: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.20
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 189: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.21
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 190: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.22
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 191: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.23
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 192: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.24
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:39 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 193: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.25
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 194: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.26
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 195: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.27
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 196: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.28
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 197: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.29
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 198: std::slice::Iter::<'_, i32>::post_inc_start.pointer_dereference.30
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:108:32 in function std::slice::Iter::<'_, i32>::post_inc_start

Check 199: std::fmt::Arguments::<'_>::new_const.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:325:34 in function std::fmt::Arguments::<'_>::new_const

Check 200: std::ptr::const_ptr::<impl *const T>::sub_ptr::runtime::<i32>.pointer.1
- Status: SUCCESS
- Description: "same object violation"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/const_ptr.rs:815:58 in function std::ptr::const_ptr::<impl *const T>::sub_ptr::runtime::<i32>

Check 201: pointer.1
- Status: SUCCESS
- Description: "same object violation"

Check 202: main::{closure#2}.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 203: main::{closure#2}.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 204: main::{closure#2}.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 205: main::{closure#2}.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 206: main::{closure#2}.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 207: main::{closure#2}.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 208: main::{closure#2}.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 209: main::{closure#2}.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 210: main::{closure#2}.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 211: main::{closure#2}.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 212: main::{closure#2}.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 213: main::{closure#2}.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:67 in function main::{closure#2}

Check 214: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 215: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 216: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 217: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 218: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 219: <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/exact_size.rs:156:9 in function <&mut std::vec::IntoIter<i32> as std::iter::ExactSizeIterator>::len

Check 220: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 221: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 222: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 223: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 224: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 225: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 226: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 227: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 228: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 229: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 230: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 231: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 232: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 233: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 234: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 235: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 236: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 237: <usize as std::option::SpecOptionPartialEq>::eq.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:2160:15 in function <usize as std::option::SpecOptionPartialEq>::eq

Check 238: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 239: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 240: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 241: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 242: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 243: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:33:33 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 244: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 245: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 246: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 247: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 248: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 249: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:29:24 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 250: <std::slice::Iter<'_, i32> as std::iter::Iterator>::next.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:163:25 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::next

Check 251: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:53: 15:58}>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:301:27 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:53: 15:58}>

Check 252: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:53: 15:58}>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:306:17 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:53: 15:58}>

Check 253: std::alloc::Layout::size.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 254: std::alloc::Layout::size.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 255: std::alloc::Layout::size.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 256: std::alloc::Layout::size.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 257: std::alloc::Layout::size.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 258: std::alloc::Layout::size.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:129:9 in function std::alloc::Layout::size

Check 259: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 260: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 261: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 262: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 263: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 264: std::ptr::NonNull::<i32>::as_ref::<'_>.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/non_null.rs:399:20 in function std::ptr::NonNull::<i32>::as_ref::<'_>

Check 265: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:64: 24:69}>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:301:27 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:64: 24:69}>

Check 266: <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:64: 24:69}>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/slice/iter/macros.rs:306:17 in function <std::slice::Iter<'_, i32> as std::iter::Iterator>::find::<{closure@tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:24:64: 24:69}>

Check 267: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 268: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 269: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 270: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 271: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 272: std::boxed::Box::<[i32]>::leak::<'_>.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:1186:24 in function std::boxed::Box::<[i32]>::leak::<'_>

Check 273: std::alloc::Layout::align.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 274: std::alloc::Layout::align.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 275: std::alloc::Layout::align.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 276: std::alloc::Layout::align.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 277: std::alloc::Layout::align.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 278: std::alloc::Layout::align.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/alloc/layout.rs:142:9 in function std::alloc::Layout::align

Check 279: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 280: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 281: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 282: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 283: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 284: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:52 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 285: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 286: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 287: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 288: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 289: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 290: std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/cmp.rs:1433:63 in function std::cmp::impls::<impl std::cmp::PartialEq for usize>::eq

Check 291: std::vec::Vec::<i32>::len.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 292: std::vec::Vec::<i32>::len.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 293: std::vec::Vec::<i32>::len.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 294: std::vec::Vec::<i32>::len.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 295: std::vec::Vec::<i32>::len.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 296: std::vec::Vec::<i32>::len.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2142:9 in function std::vec::Vec::<i32>::len

Check 297: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 298: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 299: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 300: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 301: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 302: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:409:52 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 303: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 304: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 305: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 306: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 307: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 308: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 309: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 310: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 311: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 312: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 313: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 314: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:55 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 315: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.19
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 316: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.20
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 317: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.21
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 318: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.22
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 319: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.23
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 320: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.24
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 321: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.25
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 322: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.26
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 323: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.27
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 324: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.28
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 325: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.29
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 326: <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop.pointer_dereference.30
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/into_iter.rs:411:76 in function <<std::vec::IntoIter<T, A> as std::ops::Drop>::drop::DropGuard<'_, i32, std::alloc::Global> as std::ops::Drop>::drop

Check 327: main::{closure#3}.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 328: main::{closure#3}.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 329: main::{closure#3}.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 330: main::{closure#3}.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 331: main::{closure#3}.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 332: main::{closure#3}.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 333: main::{closure#3}.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 334: main::{closure#3}.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 335: main::{closure#3}.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 336: main::{closure#3}.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 337: main::{closure#3}.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 338: main::{closure#3}.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:67 in function main::{closure#3}

Check 339: std::ptr::read::<i32>.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 340: std::ptr::read::<i32>.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 341: std::ptr::read::<i32>.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 342: std::ptr::read::<i32>.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 343: std::ptr::read::<i32>.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 344: std::ptr::read::<i32>.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ptr/mod.rs:1215:9 in function std::ptr::read::<i32>

Check 345: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 346: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 347: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 348: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 349: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 350: alloc::raw_vec::RawVec::<i32>::capacity.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:231:44 in function alloc::raw_vec::RawVec::<i32>::capacity

Check 351: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 352: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 353: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 354: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 355: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 356: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:240:25 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 357: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 358: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 359: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 360: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 361: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 362: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:250:62 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 363: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.13
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 364: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.14
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 365: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.15
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 366: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.16
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 367: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.17
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 368: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.18
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:252:23 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 369: alloc::raw_vec::RawVec::<i32>::current_memory.pointer_dereference.19
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/raw_vec.rs:241:13 in function alloc::raw_vec::RawVec::<i32>::current_memory

Check 370: main::{closure#0}.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 371: main::{closure#0}.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 372: main::{closure#0}.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 373: main::{closure#0}.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 374: main::{closure#0}.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 375: main::{closure#0}.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 376: main::{closure#0}.pointer_dereference.7
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 377: main::{closure#0}.pointer_dereference.8
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 378: main::{closure#0}.pointer_dereference.9
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 379: main::{closure#0}.pointer_dereference.10
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 380: main::{closure#0}.pointer_dereference.11
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 381: main::{closure#0}.pointer_dereference.12
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:15:56 in function main::{closure#0}

Check 382: main::{closure#1}.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 383: main::{closure#1}.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 384: main::{closure#1}.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 385: main::{closure#1}.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 386: main::{closure#1}.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 387: main::{closure#1}.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:18:56 in function main::{closure#1}

Check 388: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.1
- Status: SUCCESS
- Description: "dereference failure: pointer NULL"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref

Check 389: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.2
- Status: SUCCESS
- Description: "dereference failure: pointer invalid"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref

Check 390: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.3
- Status: SUCCESS
- Description: "dereference failure: deallocated dynamic object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref

Check 391: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.4
- Status: SUCCESS
- Description: "dereference failure: dead object"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref

Check 392: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.5
- Status: SUCCESS
- Description: "dereference failure: pointer outside object bounds"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref

Check 393: <std::vec::Vec<i32> as std::ops::Deref>::deref.pointer_dereference.6
- Status: SUCCESS
- Description: "dereference failure: invalid integer address"
- Location: ../../../.rustup/toolchains/nightly-2023-12-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/vec/mod.rs:2705:55 in function <std::vec::Vec<i32> as std::ops::Deref>::deref


SUMMARY:
** 0 of 393 failed (1 unreachable)

VERIFICATION:- SUCCESSFUL
Verification Time: 2.2231767s

Complete - 1 successfully verified harnesses, 0 failures, 1 total.
              </xmp>
            </div><!-- class="output-box" -->
            
          

          
          <div class="output-box"><xmp>stderr:
            
warning: this method call resolves to `<&[T; N] as IntoIterator>::into_iter` (due to backwards compatibility), but will resolve to <[T; N] as IntoIterator>::into_iter in Rust 2021
--> tests/bookrunner/books/Rust by Example/Functions/Closures/Examples in std/Searching through iterators/22.rs:26:47
|
26 |     println!("Find 2 in array2: {:?}", array2.into_iter().find(|&&x| x == 2));
|                                               ^^^^^^^^^
|
= warning: this changes meaning in Rust 2021
= note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/IntoIterator-for-arrays.html>
= note: `#[warn(array_into_iter)]` on by default
help: use `.iter()` instead of `.into_iter()` to avoid ambiguity
|
26 |     println!("Find 2 in array2: {:?}", array2.iter().find(|&&x| x == 2));
|                                               ~~~~
help: or use `IntoIterator::into_iter(..)` instead of `.into_iter()` to explicitly iterate by value
|
26 |     println!("Find 2 in array2: {:?}", IntoIterator::into_iter(array2).find(|&&x| x == 2));
|                                        ++++++++++++++++++++++++      ~

warning: Found the following unsupported constructs:
- caller_location (1)
- foreign function (2)

Verification will fail if one or more of these constructs is reachable.
See https://model-checking.github.io/kani/rust-feature-support.html for more details.

warning: 2 warnings emitted

            </xmp>
          </div><!-- class="output-box" -->
          

        </div><!-- class="command-content" -->
      </div><!-- class="fail" -->
    </div><!-- id="command-table" -->
    

  </div><!-- class="stage" -->
  


</div><!-- id="content" -->
</body>
</html>
